sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -889,9 +886,13 @@ private static void terminalPrettyPrint(PrintStream out, String[] words) {
           continue;
         }
       }
-
-      descriptors.add(new PropertyDescriptor(
-          propertyName, getterMethod, method));
+      // Properties can appear multiple times with subclasses, and we don't
+      // want to clobber a good entry with a bad one.
+      if (!usedDescriptors.contains(propertyName)) {
```

Comment:
`descriptors` is a list, can you say more about how this clobbers?

Is it more that we already have a mismatch so we don't need need to keep accumulating mismatches on the same property?